### ISSUE TITLE
[Announcement]:  Removal of non-essential platform services from Hosting

### ISSUE THREAD

**Issue by davidfowl** [aspnet/Hosting#514]
This is the thread to discuss https://github.com/aspnet/Announcements/issues/125

**Comment by csunali** [comment]
@davidfowl, I previously asked you if it was a good idea to use `PlatformServices.Default` static for ASP.NET apps and you said "no" in your comment at https://github.com/aspnet/dnx/issues/2990#issuecomment-152046852

Now we all end up being have to use `PlatformServices.Default` if we need to access `ILibraryManager` and `IAssemblyLoadContextAccessor`.

Do I get it wrong that we have to use `PlatformServices.Default` static (at least once during registration as you stated at https://github.com/aspnet/dnx/issues/3032#issuecomment-163691530) or if I am not wrong with what I understand, what has actually changed the teams opinion?

**Comment by davidfowl** [comment]
We're going through a transition from DNX based applications to .NET CLI based applications. These changes reflect that. Your application should be registering the static instances into the container (as needed) but your application code should be consuming the interfaces.

ILibraryManager, IAssemblyLoadContextAccessor and friends are all going away. It's likely that they will be either integrated into the BCL (like the assembly loading APIs) or a stand alone library that can be used independently.

> Do I get it wrong that we have to use PlatformServices.Default static (at least once during registration as you stated at aspnet/dnx#3032 (comment)) or if I am not wrong with what I understand, what has actually changed the teams opinion?

TL;DR yes

**Comment by csunali** [comment]
Thanks!

**Comment by davidfowl** [comment]
We'll make this a bit cleaner by exposing the implementations without them being already instantiated. e.g.

```C#
services.AddPlatformServices();
```

**Comment by csunali** [comment]
I feel like one important scenario to mention here is that; I do assembly scanning during `ConfigureServices` time. I have seen others do the same as well. The reason doing so is to call `ConfigureServices` methods on plugins that are discovered at runtime so that they can register their own services.

Long story short, `ILibraryManager`, `IAssemblyLoaderContainer` and `IAssemblyLoadContextAccessor` are trivial components for assembly scanning.

That said, even though your approach to `services.AddPlatformServices();` seems like a reasonable solution, would we be still able to access the mentioned services while still within the `ConfigureServices` method?

By the way, not within the scope of this issue but is there a way to access the `IHostingEnvironment` the same way we access the above interfaces? Again during `ConfigureServices` call. Currently we have to pass it all the way from the `Startup` class's constructor. Not a bad thing but just wondering if there is another way.

**Comment by muratg** [comment]
We are closing this issue because no further action is planned for this issue. If you still have any issues or questions, please log a new issue with any additional details that you have.
